src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CSharpWorkspaceExtensions.projitems

```diff
@@ -10,6 +10,28 @@
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="$(MSBuildThisFileDirectory)CodeActions\CSharpCodeFixOptionsProvider.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\ArgumentGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\AttributeGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\ConstructorGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\ConversionGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\CSharpCodeGenerationContextInfo.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\CSharpCodeGenerationHelpers.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\CSharpCodeGenerationService.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\CSharpDeclarationComparer.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\CSharpFlagsEnumGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\DestructorGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\EnumMemberGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\EventGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\ExpressionGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\FieldGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\MethodGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\NamedTypeGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\NamespaceGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\OperatorGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\ParameterGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\PropertyGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\StatementGenerator.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)CodeGeneration\TypeParameterGenerator.cs" />
```

Comment:
Moving all thsee ICodeGenerationService pieces for C# into shared source project.